pkg/commands/internal/migrate/migrate_linters_settings.go

```diff
@@ -0,0 +1,1017 @@
+	return versiontwo.PromlinterSettings{
+		Strict:          old.Strict,
+		DisabledLinters: old.DisabledLinters,
+	}
+}
+
+func toProtoGetterSettings(old versionone.ProtoGetterSettings) versiontwo.ProtoGetterSettings {
+	return versiontwo.ProtoGetterSettings{
+		SkipGeneratedBy:         old.SkipGeneratedBy,
+		SkipFiles:               old.SkipFiles,
+		SkipAnyGenerated:        old.SkipAnyGenerated,
+		ReplaceFirstArgInAppend: old.ReplaceFirstArgInAppend,
+	}
+}
+
+func toReassignSettings(old versionone.ReassignSettings) versiontwo.ReassignSettings {
+	return versiontwo.ReassignSettings{
+		Patterns: old.Patterns,
+	}
+}
+
+func toRecvcheckSettings(old versionone.RecvcheckSettings) versiontwo.RecvcheckSettings {
+	return versiontwo.RecvcheckSettings{
+		DisableBuiltin: old.DisableBuiltin,
+		Exclusions:     old.Exclusions,
+	}
+}
+
+func toReviveSettings(old versionone.ReviveSettings) versiontwo.ReviveSettings {
+	settings := versiontwo.ReviveSettings{
+		MaxOpenFiles:   old.MaxOpenFiles,
+		Confidence:     old.Confidence,
+		Severity:       old.Severity,
+		EnableAllRules: old.EnableAllRules,
+		ErrorCode:      old.ErrorCode,
+		WarningCode:    old.WarningCode,
+	}
+
+	for _, rule := range old.Rules {
+		settings.Rules = append(settings.Rules, versiontwo.ReviveRule{
+			Name:      rule.Name,
+			Arguments: rule.Arguments,
+			Severity:  rule.Severity,
+			Disabled:  rule.Disabled,
+			Exclude:   rule.Exclude,
+		})
+	}
+
+	for _, directive := range old.Directives {
+		settings.Directives = append(settings.Directives, versiontwo.ReviveDirective{
+			Name:     directive.Name,
+			Severity: directive.Severity,
+		})
+	}
+
+	return settings
+}
+
+func toRowsErrCheckSettings(old versionone.RowsErrCheckSettings) versiontwo.RowsErrCheckSettings {
+	return versiontwo.RowsErrCheckSettings{
+		Packages: old.Packages,
+	}
+}
+
+func toSlogLintSettings(old versionone.SlogLintSettings) versiontwo.SlogLintSettings {
+	return versiontwo.SlogLintSettings{
+		NoMixedArgs:    old.NoMixedArgs,
+		KVOnly:         old.KVOnly,
+		AttrOnly:       old.AttrOnly,
+		NoGlobal:       old.NoGlobal,
+		Context:        old.Context,
+		StaticMsg:      old.StaticMsg,
+		NoRawKeys:      old.NoRawKeys,
+		KeyNamingCase:  old.KeyNamingCase,
+		ForbiddenKeys:  old.ForbiddenKeys,
+		ArgsOnSepLines: old.ArgsOnSepLines,
+	}
+}
+
+func toSpancheckSettings(old versionone.SpancheckSettings) versiontwo.SpancheckSettings {
+	return versiontwo.SpancheckSettings{
+		Checks:                   old.Checks,
+		IgnoreCheckSignatures:    old.IgnoreCheckSignatures,
+		ExtraStartSpanSignatures: old.ExtraStartSpanSignatures,
+	}
+}
+
+func toStaticCheckSettings(old versionone.LintersSettings) versiontwo.StaticCheckSettings {
+	checks := slices.Concat(old.Staticcheck.Checks, old.Stylecheck.Checks, old.Gosimple.Checks)
+
+	slices.Sort(checks)
+
+	return versiontwo.StaticCheckSettings{
+		Checks:                  slices.Compact(checks),
```

Comment:
We can change:

```suggestion
		Checks:                  Unique(checks),
```

And remove the `slices.Sort(checks)` above.